lib/src/gregorian_calendar_duration.dart

```diff
@@ -0,0 +1,147 @@
+import 'calendar_duration.dart';
+import 'gregorian_calendar.dart';
+import 'gregorian_consts.dart';
+
+/// Represents a given duration which can be added/subtracted
+/// from a [GregorianCalendar] object
+class GregorianCalendarDuration implements CalendarDuration<GregorianCalendar>{
+
+  GregorianCalendarDuration({this.days = 0, this.weeks = 0, this.months = 0, this.years = 0});
+
+  /// Constructs a GregorianCalendarDuration where the values have been normalized
+  factory GregorianCalendarDuration.normalized({int years = 0, int months = 0, int weeks = 0, int days=0}){
+    // normalize values and then call the real constructor
+    if (days >= daysPerWeek){
+      weeks += days ~/ daysPerWeek;
+      days %= daysPerWeek;
+    }
+    else if (days <= -daysPerWeek){
+      weeks += days ~/ daysPerWeek;
+      days = -(days.abs() % daysPerWeek);
+    }
+    if (months >= monthsPerYear){
+      years += months ~/ monthsPerYear;
+      months %= monthsPerYear;
+    }
+    else if (months <= -monthsPerYear){
+      years += months ~/ monthsPerYear;
+      months = -(months.abs() % monthsPerYear);
+    }
+    return GregorianCalendarDuration(years: years, months: months, weeks: weeks, days: days);
+  }
+
+  factory GregorianCalendarDuration.fromDays(int days, GregorianCalendar referenceDate){
+    final newReferenceDate = referenceDate.addDays(days);
+    return new GregorianCalendarDuration.normalized(
+      years: newReferenceDate.year - referenceDate.year,
+      months: newReferenceDate.month - referenceDate.month,
+      days:  newReferenceDate.day - referenceDate.day
+    );
+  }
+
+  @override
+  final int days;
+
+  @override
+  final int weeks;
+
+  @override
+  final int months;
+
+  @override
+  final int years;
+
+  @override
+  GregorianCalendarDuration operator +(CalendarDuration<GregorianCalendar> other){
+    return new GregorianCalendarDuration.normalized(
+      years: years + other.years,
+      months: months + other.months,
+      weeks: weeks + other.weeks,
+      days: days + other.days,
+    );
+  }
+
+  @override
+  GregorianCalendarDuration operator -(CalendarDuration<GregorianCalendar> other){
+    return new GregorianCalendarDuration.normalized(
+      years: years - other.years,
+      months: months - other.months,
+      weeks: weeks - other.weeks,
+      days: days - other.days,
+    );
+  }
+
+  @override
+  GregorianCalendarDuration operator -(){
+    return new GregorianCalendarDuration(
+      years: -years,
+      months: -months,
+      weeks: -weeks,
+      days: -days,
+    );
+  }
+
+  @override
+  int toApproxDays(){
+    return ((years * monthsPerYear + months) * avgDaysPerMonth).round() + (weeks * daysPerWeek) + days;
```

Comment:
Why can't we just use a similar formula to what already existed?

Basically get a normalization of the date, and then get an int that is more or less ISO-8601 without the dashes?

Reply:
The problem I see with that would be in situations where the combined weeks and days exceed the length of a month or year. We can't definitively normalize days into years or months without a reference date, since years and months have a variable number of days. If we try to normalize 35 days, it could be 1 month 4 days, 1 month 5 days, 1 month 6 days, or 1 month 7 days. 

With the current normalization function, 35 days normalizes as 5 weeks. if we were to do what it sounds like you're saying, (that is, concatenate years, months, weeks, and days into a single integer) then 35 days would be something like 500, whereas 1 month would be something like 10000, meaning 35 days < 1 month. Obviously that would be wrong.

The advantage to using something like the `toApproxDays()` is that it is totally agnostic to a reference date, and still accurate to what one would typically expect. 1 year == 365 days, 1 month == 30 days, etc.

Reply:
Ahh right. I think that's a fundamental problem with adding durations then - this library is meant to make it easier to use calendar dates correctly.  Maybe the duration needs to be pegged to a year so we know how to count the days properly for the months?

Reply:
(Introducing approximations like this is bound to cause trouble for people who don't realize it's only approximate, and it can cause trouble in ways that are very hard to reproduce)

Reply:
You're right, this is a fundamental problem with the nature of the system. However, a duration is a very important real-world concept that applies in countless situations. I do feel like this is something that is needed.

I guess the only real solution is to disallow directly comparing durations to one another. Force people to either explicitly call `toDays(referenceDate` or `toApproxDays()`

Reply:
I have changed the == operator to only mark durations as equal if they are for sure exactly equal. No more approximations or averages. I have removed all other comparison operators. Now the only way durations can be compared is by explicitly converting to an integer via `toApproxDays()` (so there can be no question to the user this is an approximation) or by calling `toDays(referenceDate)` so an exact comparison can be made compared to a specific date.